packages/fxa-settings/src/pages/Signin/Signin/index.tsx

```diff
@@ -0,0 +1,201 @@
+    try {
+      // TODO: add logic to set a new password
+      // TODO: add message in Alert Bar re: success, then navigate to where appropriate
+    } catch (e) {
+      // TODO: metrics event for error
+      // TODO: Add in localized Alert Bar message for error
+      setError(e);
+    }
+  }, []);
+
+  const [error, setError] = useState('');
+  const [password, setPassword] = useState('');
+  const useLoggedIn = () => {};
+
+  const signInWithPassword = () => {};
+
+  // TODO:
+  // Add in the Alert Bar in place of the original `success` and `error` display divs
+
+  return (
+    <>
+      <div className="mb-4">
+        {isPasswordNeeded ? (
+          <FtlMsg id="signin-password-needed-header">
+            <h1 className="card-header">
+              Enter your password{' '}
+              <span className="card-subheader">for your Firefox account</span>
+            </h1>
+          </FtlMsg>
+        ) : (
+          <>
+            <FtlMsg id="signin">
+              <h1 className="card-header">Sign in</h1>
+            </FtlMsg>
+            <>
+              {serviceLogo ? (
+                <span className="card-subheader">
+                  <FtlMsg id="signin-subheader-with-logo">Continue to </FtlMsg>
+                  {serviceLogo}
+                </span>
+              ) : (
+                <FtlMsg
+                  id="signin-subheader-without-logo"
+                  vars={{ serviceName }}
+                >
+                  <span className="card-subheader">
+                    Continue to {serviceName}
+                  </span>
+                </FtlMsg>
+              )}
+            </>
+          </>
+        )}
+      </div>
+      <section>
+        {/* Alerts and success messages originally went here */}
+        <div className="mt-9">
+          <div className="avatar-wrapper mx-auto h-24 w-24 mobileLandscape:h-40 mobileLandscape:w-40"></div>
+          <div className="my-5 text-base break-all">{email}</div>
+        </div>
+        <form noValidate {...{ onSubmit }}>
+          <input type="email" className="email hidden" value={email} disabled />
+
+          {isPasswordNeeded && (
+            <InputPassword
+              className="mb-2"
+              label={localizedPasswordFormLabel}
+              hasErrors={error.length > 0}
+              errorText={error}
+              tooltipPosition="bottom"
+              required={true}
+              autoFocus={true}
+              onChange={(e) => setPassword(e.currentTarget.value)}
+            />
+          )}
+          {/* This non-fulfilled input tricks the browser, when trying to
+              sign in with the wrong password, into not showing the doorhanger.
+           */}
+          <input className="hidden" required />
+
+          <div className="flex">
+            <button
+              className="cta-primary cta-xl"
+              type="submit"
+              onClick={isPasswordNeeded ? useLoggedIn : signInWithPassword}
```

Comment:
Should the CTA be handled with onClick here, or the form's onSubmit, with conditional check moved there? And if keeping the conditional check, should it be flipped? `isPasswordNeeded ? signInWithPassword  : useLoggedIn`

Reply:
I think you're right on all counts! Thank you so much for all this feedback!!